scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/Synchronization.java

```diff
@@ -0,0 +1,1130 @@
+     * specified value.  More formally, returns <tt>true</tt> if and only if
+     * this channel map contains at least one mapping to a value <tt>v</tt> such that
+     * <tt>(value==null ? v==null : value.equals(v))</tt>.  This operation
+     * will probably require time linear in the channel map size for most
+     * implementations of the <tt>Map</tt> interface.
+     *
+     * @param channel channel identifier
+     * @param value value whose presence in this channel map is to be tested
+     * @return <tt>true</tt> if this channel map maps one or more keys to the
+     *         specified value
+     * @throws ClassCastException if the value is of an inappropriate type for
+     *         this channel map
+     * (<a href="{@docRoot}/java/util/Collection.html#optional-restrictions">optional</a>)
+     * @throws NullPointerException if the specified value is null and this
+     *         channel map does not permit null values
+     * (<a href="{@docRoot}/java/util/Collection.html#optional-restrictions">optional</a>)
+     * @throws InvalidChannelException if the channel does not exist
+     */
+    boolean containsValue(String channel, Serializable value) throws InvalidChannelException;
+
+    /**
+     * Returns the value to which the specified key is mapped,
+     * or {@code null} if this channel map contains no mapping for the key.
+     *
+     * <p>More formally, if this channel map contains a mapping from a key
+     * {@code k} to a value {@code v} such that {@code (key==null ? k==null :
+     * key.equals(k))}, then this method returns {@code v}; otherwise
+     * it returns {@code null}.  (There can be at most one such mapping.)
+     *
+     * <p>If this channel map permits null values, then a return value of
+     * {@code null} does not <i>necessarily</i> indicate that the channel map
+     * contains no mapping for the key; it's also possible that the channel map
+     * explicitly maps the key to {@code null}.  The {@link #containsKey
+     * containsKey} operation may be used to distinguish these two cases.
+     *
+     * @param channel channel identifier
+     * @param key the key whose associated value is to be returned
+     * @return the value to which the specified key is mapped, or
+     *         {@code null} if this channel map contains no mapping for the key
+     * @throws ClassCastException if the key is of an inappropriate type for
+     *         this channel map
+     * (<a href="{@docRoot}/java/util/Collection.html#optional-restrictions">optional</a>)
+     * @throws NullPointerException if the specified key is null and this channel map
+     *         does not permit null keys
+     * (<a href="{@docRoot}/java/util/Collection.html#optional-restrictions">optional</a>)
+     * @throws InvalidChannelException if the channel does not exist
+     */
+    <T extends Serializable> T get(String channel, String key) throws InvalidChannelException;
```

Comment:
I do not like that the null both means absence of value and possible value itself

Reply:
HashMap indeed can contain nulls as keys as values. Sic!  Then it is totally fine)

Reply:
you think I should add wrapper code to prevent this ?

Reply:
After giving it second thought, the problem is that the store is backed up by HashMap objects, and HashMap supports null keys and values.

Even though it would be possible to prevent null keys or values on methods like put, putIfAbsent etc. There are also lambdas which are evaluated directly by the hashmap, making it impossible to track.

One possible fix it to use ConcurrentHashMap instead of HashMap, the former prevent null keys and values.

Refer to this discussion:
https://stackoverflow.com/questions/698638/why-does-concurrenthashmap-prevent-null-keys-and-values

Reply:
Though inside this discussion, we can note that ConcurrentHashMap precisely avoid null key/values because it would break the coherence of the collection's parallel view.

In our case, disabling null values would prevent us from doing atomic conditional removals such as:

synchronizationapi.conditionalCompute(channel, key, predicate, lambda_which_returns_a_null_value)

Without null values, we would have to do

actionResult = synchronizationapi.conditionalCompute(channel, key, predicate, identity_lambda)
if (actionResult.isTrue()) {
synchronizationapi.compute(channel, key, lambda_which_returns_a_null_value)
}